packages/headless/src/features/search/search-actions.ts

```diff
@@ -142,12 +142,10 @@ export const executeSearch = createAsyncThunk<
         ...fetched,
         response: fetched.response.success,
         automaticallyCorrected: false,
-        analyticsAction,
+        analyticsActions: [analyticsAction],
       };
     }
 
-    dispatch(analyticsAction);
-
```

Comment:
The `dispatch` here is removed in favor of returning this action at the end of the function.

The issue with this approach is that, if `isErrorResponse()` returns `true`, then the `analyticsAction` is never going to be dispatched.

I *could* add another dispatch inside the body of the condition, but then the event would include the wrong results when there's an error and throw an error when there's none.

The fix for this would be to allow errors to also log analytics (and allow analytics to be logged without results), but I'm wondering: should we still dispatch analytics in case of an error?

Reply:
No I think dispatching `logQueryError` is sufficient 😄 . It's better now IMO

Reply:
> but then the event would include the wrong results when there's an error and throw an error when there's none.

Does the event include the wrong results because it is reading from state and the reducers have not run yet?
If so, could it be simpler to pass the correct results to the analytics action?

Reply:
Unless I'm wrong on how to do it, it seems like passing the results and the query id before they are set in the state would require even more plumbing than this.